core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1072,31 +1073,226 @@ class KafkaApis(val requestChannel: RequestChannel,
+          val leaderNode = getCurrentLeader(tp.topicPartition(), request.context.listenerName)
+          leaderNode.node.foreach { node =>
+            nodeEndpoints.put(node.id(), node)
+          }
+          partitionData.currentLeader()
+            .setLeaderId(leaderNode.leaderId)
+            .setLeaderEpoch(leaderNode.leaderEpoch)
+        case _ =>
+      }
+
+      partitions.put(tp, partitionData)
+    }
+    erroneous.foreach { case (tp, data) => partitions.put(tp, data) }
+
+    var unconvertedShareAcknowledgeResponse: ShareAcknowledgeResponse = null
+
+    def createResponse(): ShareAcknowledgeResponse = {
+      val convertedData = new util.LinkedHashMap[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData]
+      unconvertedShareAcknowledgeResponse.data().responses().forEach { topicResponse =>
+        topicResponse.partitions().forEach { unconvertedPartitionData =>
+          val tp = new TopicIdPartition(
+            topicResponse.topicId,
+            new TopicPartition(topicNames.get(topicResponse.topicId()), unconvertedPartitionData.partitionIndex())
+          )
+          val error = Errors.forCode(unconvertedPartitionData.errorCode)
+          if (error != Errors.NONE)
+            debug(s"Share Acknowledge request with correlation id ${request.header.correlationId} from client $clientId " +
+              s"on partition $tp failed due to ${error.exceptionName}")
+          convertedData.put(tp, unconvertedPartitionData)
+        }
+      }
+
+      // Prepare share acknowledge response from converted data
+      ShareAcknowledgeResponse.of(
+        unconvertedShareAcknowledgeResponse.error,
+        0,
+        convertedData,
+        nodeEndpoints.values.toList.asJava
+      )
+    }
+    unconvertedShareAcknowledgeResponse = new ShareAcknowledgeResponse(
+      ShareAcknowledgeResponse.toMessage(Errors.NONE, 0, partitions.entrySet().iterator(), Collections.emptyList())
+    )
+    createResponse()
+  }
+
+  def handleAcknowledgeFromShareFetchRequest (
+                                             request: RequestChannel.Request,
+                                             topicNames : util.Map[Uuid, String],
+                                             sharePartitionManager : SharePartitionManager,
+                                           ) : ShareAcknowledgeResponse = {
+
     val shareFetchRequest = request.body[ShareFetchRequest]
+    val groupId = shareFetchRequest.data().groupId()
+    val clientId = request.header.clientId
+    val memberId = shareFetchRequest.data().memberId()
 
-    if (!config.isShareGroupEnabled) {
-      // The API is not supported when the configuration `group.share.enable` has not been set
-      requestHelper.sendMaybeThrottle(request, shareFetchRequest.getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
-      CompletableFuture.completedFuture[Unit](())
-      return
+    if (!authHelper.authorize(request.context, READ, GROUP, groupId)) {
```

Comment:
I would have expected all of the auth checking to be at the start of `handleShareFetchRequest`.